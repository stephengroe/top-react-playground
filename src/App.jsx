import { useState } from 'react'
import './App.css'
import Greeting from './Greeting'
import Person from './Person'
import Clock from './Clock'

function App() {
  const [heading, setHeading] = useState('Greetings!');

  const clickHandler = () => {
    setHeading('Salutations!');
  }

  return (
    <>
    <button type='button' onClick={clickHandler}>
      Click me!
    </button>
    <h1>{heading}</h1>
    </>
  )
}

export default App
