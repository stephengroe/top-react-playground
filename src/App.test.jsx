import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App component", () => {
  it("renders initial heading", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("renders new heading on click", async () => {
    const user = userEvent.setup();

    render(<App />);
    const button = screen.getByRole("button", { name: 'Click me!'});

    await user.click(button);
    expect(screen.getByRole("heading").textContent).toMatch(/Salutations/i)
  })
});