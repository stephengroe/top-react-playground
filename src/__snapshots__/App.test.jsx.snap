// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders initial heading 1`] = `
<div>
  <button
    type="button"
  >
    Click me!
  </button>
  <h1>
    Greetings!
  </h1>
</div>
`;
